### PR TITLE
Feat/energy consumption

### PR DESCRIPTION
**Pull request by xiaoChannel**
Android performance - power usage

### CODE REVIEW

File: Android/dokit/src/main/java/com/didichuxing/doraemonkit/kit/webdoor/bean/FpsJsbridgeBean.kt [hunk 1]
```diff
@@ -1,28 +1,56 @@
 package com.didichuxing.doraemonkit.kit.webdoor.bean
 
+import android.os.Build
+import androidx.annotation.RequiresApi
 import com.didichuxing.doraemonkit.database.FpsEntity
+import com.didichuxing.doraemonkit.database.MemoryEntity
 
 data class FpsJsbridgeBean(
```
Comment (seazon) [inline Android/dokit/src/main/java/com/didichuxing/doraemonkit/kit/webdoor/bean/FpsJsbridgeBean.kt]: Class name is inconsistent with the fields. Why add network, launch time, memory, location data here?

File: Android/dokit/src/main/java/com/didichuxing/doraemonkit/kit/webdoor/bean/NetWorkBean.kt [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+    val slowRequestCount: Int,
+    val summaryRequestCount: Int,
+    val summaryRequestDownFlow: String,
+    val summaryRequestTime: Double,
+    val summaryRequestUploadFlow: String,
+    val uploadDataRank: List<UploadDataRank>
+)
+
+fun convertToNetWorkFrom(list: List<NetworkRecordDBEntity>): NetWorkBean {
+    val downloadDataRank = ArrayList<DownloadDataRank>(5)
```
Comment (seazon) [inline Android/dokit/src/main/java/com/didichuxing/doraemonkit/kit/webdoor/bean/NetWorkBean.kt]: magic number `5`, should replace with a const value

File: Android/dokit/src/main/java/com/didichuxing/doraemonkit/kit/webdoor/bean/NetWorkBean.kt [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+                    it.value = it.value + (net.endTime - net.startTime)
+                }
+            }
+        } else {
+            if (reqTimeRank.size < 5) {
+                reqTimeRank.add(ReqTimeRank(net.url, (net.endTime - net.startTime)))
+            }
+        }
+
+        if (((net.startTime - net.endTime) / 1000 / 60 / 60) > 1000) {
```
Comment (seazon) [inline Android/dokit/src/main/java/com/didichuxing/doraemonkit/kit/webdoor/bean/NetWorkBean.kt]: should replace with a const value, eg: 
```
const val ONE_HOUR_IN_MILLISECOND = 1000 * 60 * 60
```

File: Android/dokit/src/main/res/layout/dk_fragment_web_door.xml [hunk 4]
```diff
@@ -26,6 +26,7 @@
             android:layout_marginTop="15dp"
             android:layout_marginRight="15dp"
             android:gravity="top"
+            android:text="https://trailsquad.github.io/wiredexam-react-app/"
```
Comment (seazon) [inline Android/dokit/src/main/res/layout/dk_fragment_web_door.xml]: it should be a configurable value, let's move to strings.xml first.

**Review by seazon — CHANGES_REQUESTED** [review]

**Comment by xiaoChannel** [comment]
has solved relevant suggestions, thank you

File: Android/dokit/src/main/java/com/didichuxing/doraemonkit/kit/webdoor/bean/FpsJsbridgeBean.kt (hunk 1 above)
Comment (xiaoChannel) [reply Android/dokit/src/main/java/com/didichuxing/doraemonkit/kit/webdoor/bean/FpsJsbridgeBean.kt]: has solved relevant suggestions, thank you

File: Android/dokit/src/main/java/com/didichuxing/doraemonkit/kit/webdoor/WebDoorDefaultFragment.java [hunk 5]
```diff
@@ -72,13 +73,13 @@ public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceStat
             button.setOnClickListener(new View.OnClickListener() {
                 @Override
                 public void onClick(View view) {
-                    FpsJsbridgeBean fpsJsBridgeBean = null;
+                    JsbridgeBean fpsJsBridgeBean = null;
                     if (android.os.Build.VERSION.SDK_INT >= android.os.Build.VERSION_CODES.N) {
```
Comment (seazon) [inline Android/dokit/src/main/java/com/didichuxing/doraemonkit/kit/webdoor/WebDoorDefaultFragment.java]: no data if os version > N?

File: Android/dokit/src/main/java/com/didichuxing/doraemonkit/kit/webdoor/bean/NetWorkBean.kt [hunk 6]
```diff
@@ -18,30 +18,32 @@ data class NetWorkBean(
 )
 
 fun convertToNetWorkFrom(list: List<NetworkRecordDBEntity>): NetWorkBean {
-    val downloadDataRank = ArrayList<DownloadDataRank>(5)
-    val failReqCountRank = ArrayList<FailReqCountRank>(5)
-    val reqCountRank = ArrayList<ReqCountRank>(5)
-    val reqTimeRank = ArrayList<ReqTimeRank>(5)
+    val maxSize = 5
```
Comment (seazon) [inline Android/dokit/src/main/java/com/didichuxing/doraemonkit/kit/webdoor/bean/NetWorkBean.kt]: should be `const`

File: Android/dokit/src/main/java/com/didichuxing/doraemonkit/kit/webdoor/bean/NetWorkBean.kt [hunk 7]
```diff
@@ -18,30 +18,32 @@ data class NetWorkBean(
...
+    val limiTime = 1000 * 60 * 60
```
Comment (seazon) [inline Android/dokit/src/main/java/com/didichuxing/doraemonkit/kit/webdoor/bean/NetWorkBean.kt]: should be `const`

File: Android/dokit/src/main/java/com/didichuxing/doraemonkit/kit/webdoor/bean/NetWorkBean.kt [hunk 8]
```diff
@@ -63,12 +65,12 @@ fun convertToNetWorkFrom(list: List<NetworkRecordDBEntity>): NetWorkBean {
...
             }
         } else {
-            if (reqTimeRank.size < 5) {
+            if (reqTimeRank.size < maxSize) {
                 reqTimeRank.add(ReqTimeRank(net.url, (net.endTime - net.startTime)))
             }
         }
 
-        if (((net.startTime - net.endTime) / 1000 / 60 / 60) > 1000) {
+        if ((net.startTime - net.endTime) > limiTime) {
```
Comment (seazon) [inline Android/dokit/src/main/java/com/didichuxing/doraemonkit/kit/webdoor/bean/NetWorkBean.kt]: compare with the original code, the logic changed. which is correct?

**Review by seazon — CHANGES_REQUESTED** [review]

**Review by seazon — APPROVED** [review]

**Pull request merged**